tests/test_crf.py

```diff
@@ -467,3 +467,130 @@ def test_first_timestep_mask_is_not_all_on(self):
+                                                             f"not matching non-scripted forward output {llh_no_mask}"
+        # No mask means the mask is all ones
+        llh_mask = crf(emissions, tags, mask=torch.ones_like(tags).byte())
+        llh_mask_script = crf_script(emissions, tags, mask=torch.ones_like(tags).byte())
+        assert torch.equal(llh_mask, llh_mask_script), f"scripted crf forward output {llh_mask_script} " \
+                                                       f"not matching non-scripted forward output {llh_mask}"
+
+        # Test scripted forward in batched setting
+        batch_size = 10
+        # shape: (seq_length, batch_size, num_tags)
+        emissions_batch = make_emissions(crf, batch_size=batch_size)
+        # shape: (seq_length, batch_size)
+        tags_batch = make_tags(crf, batch_size=batch_size)
+        llh = crf(emissions_batch, tags_batch)
+        llh_script = crf_script(emissions_batch, tags_batch)
+        assert torch.equal(llh_script, llh), f"scripted crf forward output {llh_script} " \
+                                             f"not matching non-scripted forward output {llh}"
+
+        # Test scripted forward when reduction is None, mean, token_mean
+        # shape: (seq_length, batch_size, num_tags)
+        emissions = make_emissions(crf)
+        # shape: (seq_length, batch_size)
+        tags = make_tags(crf)
+        llh = crf(emissions, tags, reduction='none')
+        llh_script = crf_script(emissions, tags, reduction='none')
+        assert torch.equal(llh_script, llh), f"scripted crf forward output {llh_script} " \
+                                             f"not matching non-scripted forward output {llh}"
+        llh = crf(emissions, tags, reduction='mean')
+        llh_script = crf_script(emissions, tags, reduction='mean')
+        assert torch.equal(llh_script, llh), f"scripted crf forward output {llh_script} " \
+                                             f"not matching non-scripted forward output {llh}"
+
+        mask = torch.tensor([[1, 1, 1], [1, 1, 0]], dtype=torch.uint8).transpose(0, 1)
+        llh = crf(emissions, tags, mask=mask, reduction='token_mean')
+        llh_script = crf_script(emissions, tags, mask=mask, reduction='token_mean')
+        assert torch.equal(llh_script, llh), f"scripted crf forward output {llh_script} " \
+                                             f"not matching non-scripted forward output {llh}"
+
+        # Test scripted forward when running batch first mode
+        crf_bf = make_crf(batch_first=True)
+        # Copy parameter values from non-batch-first CRF; requires_grad must be False
+        # to avoid runtime error of in-place operation on a leaf variable
+        crf_bf.start_transitions.requires_grad_(False).copy_(crf.start_transitions)
+        crf_bf.end_transitions.requires_grad_(False).copy_(crf.end_transitions)
+        crf_bf.transitions.requires_grad_(False).copy_(crf.transitions)
```

Comment:
Why do you need to copy the parameters? Same question for similar changes in `test_scripted_decode`.

Reply:
I believe these were copied from the original test code. These copying statements are also present in the tests above, e.g. https://github.com/kmkurn/pytorch-crf/blob/c885fba8b3754eb20520166c75ae2a50ae99847d/tests/test_crf.py#L272-L276.